examples/chip-tool/py_matter_chip_tool_adapter/matter_chip_tool_adapter/decoder.py

```diff
@@ -348,7 +385,7 @@ def run(self, specs, value, cluster_name: str, typename: str, array: bool):
                 del value[key_name]
 
         elif isinstance(value, list) and array:
-            value = [self.run(specs, v, cluster_name, typename, False)
+            value = [self.run(specs, v, cluster_name, typename, isinstance(v, list))
```

Comment:
Hmm, why is this needed?  Are there lists of lists in the spec somewhere?

Reply:
> Hmm, why is this needed? Are there lists of lists in the spec somewhere?

Not really. This is mainly due to the return format used for API return values such as [1].

For instance, if you run the following command:

```
$ darwin-framework-tool any read-by-id 29 0 0x12344321 65535
```

The return value will be:

```
[[{'DeviceType': 17, 'Revision': 1}, {'DeviceType': 22, 'Revision': 1}], [{'DeviceType': 256, 'Revision': 1}, {'DeviceType': 17, 'Revision': 1}], [{'DeviceType': 256, 'Revision': 1}, {'DeviceType': 17, 'Revision': 1}], [{'DeviceType': 15, 'Revision': 3}], [{'DeviceType': 15, 'Revision': 3}]]
```

This format returns a list of lists containing dictionaries.

[1] https://github.com/project-chip/connectedhomeip/blob/master/examples/darwin-framework-tool/commands/clusters/ReportCommandBridge.h#L60

Reply:
Ah, OK.  Might be worth documenting that.